lib/Bio/EnsEMBL/DataCheck/Checks/CheckIDRangeMLSS.pm

```diff
@@ -0,0 +1,131 @@
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['compara'],
+  TABLES         => ['genomic_align', 'genomic_align_block', 'genomic_align_tree', 'constrained_element', 'conservation_score', 'dnafrag']
+};
+
+
+sub skip_tests {
+  my ($self) = @_;
+  if (is_compara_ehive_db($self->dba) == 1) {
+    return( 1, "This check is not intended for pipeline databases" );
+  }
+}
+
+
+sub tests {
+  my ($self) = @_;
+
+  my $desc_1 = "genomic_align.genomic_align_id all within correct MLSS range";
+  my $sql_1 = q/
+    SELECT genomic_align_id, method_link_species_set_id
+    FROM genomic_align
+    WHERE genomic_align_id < (method_link_species_set_id * 10000000000) 
+      OR genomic_align_id > ((method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_1, $desc_1);
+
+
+  my $desc_2 = "genomic_align_block.genomic_align_block_id all within correct MLSS range";
+  my $sql_2 = q/
+    SELECT genomic_align_block_id, method_link_species_set_id
+    FROM genomic_align_block 
+    WHERE genomic_align_block_id < (method_link_species_set_id * 10000000000) 
+      OR genomic_align_block_id > ((method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_2, $desc_2);
+
+
+  my $desc_3 = "genomic_align_block.group_id all within correct MLSS range";
+  my $sql_3 = q/
+    SELECT group_id, method_link_species_set_id
+    FROM genomic_align_block 
+    WHERE group_id < (method_link_species_set_id * 10000000000) 
+      OR group_id > ((method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_3, $desc_3);
+
+
+  my $desc_4 = "genomic_align_tree.node_id all within correct MLSS range";
+  my $sql_4 = q/
+    SELECT DISTINCT(t.node_id), g.method_link_species_set_id 
+    FROM genomic_align_tree t 
+    JOIN genomic_align g 
+      ON g.node_id=t.node_id
+    WHERE t.node_id < (g.method_link_species_set_id * 10000000000) 
+      OR t.node_id > ((g.method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_4, $desc_4);
+
+
+  my $desc_5 = "constrained_element.constrained_element_id all within correct MLSS range";
+  my $sql_5 = q/
+    SELECT constrained_element_id, method_link_species_set_id
+    FROM constrained_element 
+    WHERE constrained_element_id < (method_link_species_set_id * 10000000000) 
+      OR constrained_element_id > ((method_link_species_set_id + 1) * 10000000000)
+  /;
+  is_rows_zero($self->dba, $sql_5, $desc_5);
+
+
+  my $desc_6 = "dnafrag.dnafrag_id for ancestral frags all within correct MLSS range";
+  my $sql_6 = q/
+    SELECT d.dnafrag_id, g.method_link_species_set_id 
+    FROM dnafrag d 
+    JOIN genomic_align g 
+      ON g.dnafrag_id=d.dnafrag_id
+    WHERE d.name LIKE 'Ancestor%'
+      AND (
+        d.dnafrag_id < (g.method_link_species_set_id * 10000000000) 
+        OR d.dnafrag_id > ((g.method_link_species_set_id + 1) * 10000000000)
+      )
```

Comment:
Just as a curiosity, if you wanted to avoid the `JOIN` (I have tested and it is not that slow, so it is not worth to change it, I think), you could have used `SUBSTRING_INDEX(SUBSTRING_INDEX(d.name, '_', 2), '_', -1)` to get the MLSS ID from the `dnafrag` table.

Reply:
I assumed the `name` is datachecked elsewhere,  but where possible I always think it is best to go to the source when comparing data such as IDs.